src/core/crypto.rs

```diff
@@ -74,3 +94,21 @@ pub fn verify_rsa_signature(
         .verify(params, msg, signature)
         .map_err(|_| SignatureVerificationError::CryptoError("bad signature".to_string()))
 }
+
+pub fn verify_ec_signature(
+    key: &CoreJsonWebKey,
+    params: &'static ring_signature::EcdsaVerificationAlgorithm,
+    msg: &[u8],
+    signature: &[u8],
+) -> Result<(), SignatureVerificationError> {
+    let (x, y, crv) = ec_public_key(&key).map_err(SignatureVerificationError::InvalidKey)?;
+    if *crv == CoreJsonCurveType::P521{
+        return Err(SignatureVerificationError::UnsupportedAlg("Only P256 and P384 are supported for now".to_string()));
+    }
+    let mut pk = vec![0x04];
```

Comment:
mind adding a comment linking to the relevant RFC/doc that defines this key param concatenation?